### PR TITLE
[SDK-217] Allow consumers to set app name and channel for experiment filtering

### PR DESCRIPTION
**Pull request by travis79**
This adds the ability for the consuming app to set the `appName` and `channel` fields for filtering purposes in Nimbus experiments.  

This also updates the samples-glean Nimbus integration with these changes, along with changing it to use the `feature-id` rather than the `experiment-slug` for fetching branches.

### CODE REVIEW

**Comment by travis79** [comment]
This shouldn't land until [this](https://github.com/mozilla/nimbus-sdk/pull/97) lands and we have a new release of Application Services with it in there.

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ object Versions {
     const val disklrucache = "2.0.2"
     const val leakcanary = "2.4"
 
-    const val mozilla_appservices = "72.1.0"
+    const val mozilla_appservices = "72.1.0-TESTING"
```
Comment (travis79) [inline buildSrc/src/main/java/Dependencies.kt:33]: Change this to the appservices release with the Nimbus changes from [here](https://github.com/mozilla/nimbus-sdk/pull/97) in it.

File: samples/glean/src/main/java/org/mozilla/samples/glean/MainActivity.kt [hunk 2]
```diff
@@ -101,7 +101,7 @@ open class MainActivity : AppCompatActivity(), NimbusApi.Observer {
     @MainThread
     private fun configureButton() {
         val nimbus = GleanApplication.nimbus
-        val branch = nimbus.getExperimentBranch("test-color")
+        val branch = nimbus.getExperimentBranch("sample-experiment-feature")
```
Comment (travis79) [inline samples/glean/src/main/java/org/mozilla/samples/glean/MainActivity.kt:104]: Requesting the branch by feature-id seems to work well!

File: samples/glean/src/main/res/raw/initial_experiments.json [hunk 3]
```diff
@@ -2,36 +2,57 @@
...
     "endDate": null,
-    "branches": [{
-      "slug": "control",
-      "ratio": 1
-    },
+    "branches": [
+      {
+        "slug": "control",
+        "ratio": 1,
+        "feature": {
```
Comment (travis79) [inline samples/glean/src/main/res/raw/initial_experiments.json:9]: The `feature` embedded in the branch seems so redundant...

**Review by travis79 — COMMENTED** [review]
Added a few review comments as notes.

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt [hunk 4]
```diff
@@ -180,8 +180,10 @@ private val loggingErrorReporter: ErrorReporter = { e: Throwable ->
 /**
  * A implementation of the [NimbusApi] interface backed by the Nimbus SDK.
  */
-@Suppress("LargeClass")
+@Suppress("LargeClass", "LongParameterList")
 class Nimbus(
+    private val appName: String,
+    private val channel: String,
```
Comment (jhugman) [inline components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:186]: By convention,`Context` is the first argument.

Also: I think I'd suggest putting the channel and app name in a small `data class` object, `NimbusAppSettings`.

File: components/browser/engine-gecko/docs/metrics.md [hunk 5]
```diff
@@ -22,9 +22,8 @@ In addition to those built-in metrics, the following metrics are added to the pi
 
 | Name | Type | Description | Data reviews | Extras | Expiration | [Data Sensitivity](https://wiki.mozilla.org/Firefox/Data_Collection) |
 | --- | --- | --- | --- | --- | --- | --- |
-| avif.aom_decode_error |[quantity](https://mozilla.github.io/glean/book/user/metrics/quantity.html) |Image-decode Error from AOM decoder  |[Bug 1670827](https://bugzilla.mozilla.org/show_bug.cgi?id=1670827#c9)|<ul><li>unit: error code</li></ul>|never | |
+| avif.aom_decode_error |[labeled_counter](https://mozilla.github.io/glean/book/user/metrics/labeled_counters.html) |Image-decode Error from AOM decoder  |[Bug 1690406](https://bugzilla.mozilla.org/show_bug.cgi?id=1690406#c3)|<ul><li>error</li><li>mem_error</li><li>abi_mismatch</li><li>incapable</li><li>unsup_bitstream</li><li>unsup_feature</li><li>corrupt_frame</li><li>invalid_param</li></ul>|never | |
```
Comment (jhugman) [inline components/browser/engine-gecko/docs/metrics.md:25]: Should we be including these metrics changes in unrelated feature PRs? I've come across this before, but have I been doing it wrong?

**Review by jhugman — APPROVED** [review]
A couple of suggested changes, but nothing that should be held up by another round of review. 🎉

**Comment by st3fan** [comment]
bors try

File: components/browser/engine-gecko/docs/metrics.md (hunk 5 above)
Comment (travis79) [reply components/browser/engine-gecko/docs/metrics.md:25]: There was an update to Glean, which may have changed how the docs were generated, which is the only reason I included this one.

**Comment by travis79** [comment]
Okay, review comments addressed...   fingers crossed for <green>CI</green>

**Comment by st3fan** [comment]
bors try

**Pull request merged**